sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HintErrorLogger.scala

```diff
@@ -42,6 +45,17 @@ object HintErrorLogger extends HintErrorHandler with Logging {
     logWarning(s"A join hint $hint is specified but it is not part of a join relation.")
   }
 
+  override def joinBuildSideNotSupported(joinType: JoinType, joinHint: JoinHint): Unit = {
+    assert(joinHint.leftHint.isDefined || joinHint.rightHint.isDefined)
+    val (hint, buildSide) = if (joinHint.leftHint.isDefined) {
+      (joinHint.leftHint.get, "left")
+    } else {
+      (joinHint.rightHint.get, "right")
+    }
+    logWarning(s"A join hint $hint is specified but it is not supported with build $buildSide " +
```

Comment:
nit: `A join hint $hint ` => `Hint $hint` to follow the existing msg: https://github.com/apache/spark/pull/32355/files#diff-afe669f38210f85c87450ace74311b4ce2301e6aeab820aea444fcd802c2d5c2R60

Reply:
This msg format and method follow that `joinNotFoundForJoinHint` msg. Seems that one is better ?